x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/AuthorizationService.java

```diff
@@ -262,6 +264,7 @@ public void authorize(
     ) {
 
         final AuthorizationContext enclosingContext = extractAuthorizationContext(threadContext, action);
+        final ParentActionAuthorization parentAuthorization = extractParentAuthorization(threadContext);
 
```

Comment:
Parent authz on the remote node is conveyed in a regular header rather than a transient one, so there is no need to store it like this because it is not cleared by the `threadContext.newStoredContext` call below (line 276).

I think this oversight is causing the multi-cluster search failures https://gradle-enterprise.elastic.co/s/p6jiurxwagew6 . The "permanent" header is set and is also carried over remote connections.

Reply:
> Parent authz on the remote node is conveyed in a regular header rather than a transient one, so there is no need to store it like this because it is not cleared by the

We talked it over in a sync meeting.
We concluded that it's preferable from a code structure point of view to access (in this case read) the threadContext in the `AuthorizationService` rather than the `AuthorizationEngine` (`RBACEngine`).